src/components/DatePicker/DatePicker.vue

```diff
@@ -28,6 +28,7 @@
                         :disabled="disablePicker"
                         :hint="hint"
                         :persistent-hint="persistentHint"
+                        @click:clear="emitCancel()"
```

Comment:
Do you want this to emit cancel and close the dialog? Or only clear the date?

Reply:
> Please rebuild Storybook and check those files in.

Sounds good. What does that do if there isn't a new storybook file? Is it to ensure the existing storybook suites contain the new code?

Reply:
for my case I just want it to emit cancel so that it triggers the same flow

Reply:
otherwise I'll need to add another emit that would be used to trigger the same flow (I thought this was unnecessary)

Reply:
i think we should add a separate emit for the clear.  That way we don't run into any undesired side effects if emitCancel is ever changed.

Reply:
So if the user clicks Cancel then it clear the date? I think that might cause some users to accidentally clear their date (thinking that Cancel would just close the dialog). Could you ask Tracey for her thoughts on this?

Reply:
Note: this isn't emitting a "clear" event. Clearing the dialog is a local function.

Reply:
Oh I get it... this is the "X" on the text field that is used to clear the input.

Reply:
To self-document this (since a couple of us were obviously confused), I recomment adding a new click event function (eg, onClickClear) that, in turn, calls emitCancel.

Reply:
😢

Reply:
'It's a one line change it'll be 2 secs' - famous last words

Reply:
If I'm going back in there I will add a new emit fn in case anyone in the future has a separate flow for the cancel vs clear

Reply:
Do we like the current change? New emit + clear ?

Reply:
i'm good with this.